shared/src/main/scala/mlscript/ucs/stages/Normalization.scala

```diff
@@ -197,57 +205,117 @@ trait Normalization { self: Desugarer with Traceable =>
+                continuation :++ specializeImpl(tail)
+              } else if (thatPattern <:< pattern) {
+                println(s"Case 1.1.2: $pattern <:< $thatPattern")
+                pattern.markAsRefined; split
+              } else {
+                println(s"Case 1.1.3: $pattern is unrelated with $thatPattern")
+                // The `continuation` is discarded because `thatPattern` is unrelated
+                // to the specialization topic.
+                if (!split.isFallback) {
+                  println(s"report warning")
+                  if (pattern <:< thatPattern) {
+                    raiseDesugaringWarning(
+                      msg"the pattern always matches" -> thatPattern.toLoc,
+                      msg"the scrutinee was matched against ${pattern.toString}" -> pattern.toLoc,
+                      msg"which is a subtype of ${thatPattern.toString}" -> (pattern match {
+                        case Pattern.Class(_, symbol, _) => symbol.defn.toLoc
+                        case _ => thatPattern.getLoc
+                      }))
+                    continuation :++ specializeImpl(tail)
+                  } else {
+                    raiseDesugaringWarning(
+                      msg"possibly conflicted patterns" -> thatPattern.toLoc,
+                      msg"the scrutinee was matched against ${pattern.toString}" -> pattern.toLoc,
+                      msg"which is unrelated with ${thatPattern.toString}" -> thatPattern.toLoc)
+                    specializeImpl(tail)
+                  }
+                } else {
+                  specializeImpl(tail)
+                }
+              }
+            case -(full) =>
+              println(s"Case 1.2: ${scrutineeVar.name} === ${thatScrutineeVar.name}")
+              thatPattern reportInconsistentRefinedWith pattern
+              val samePattern = thatPattern =:= pattern
+              if (samePattern || thatPattern <:< pattern) {
+                println(s"Case 1.2.1: $pattern =:= (or <:<) $thatPattern")
+                // The `continuation` is discarded because `thatPattern` is related
+                // to the specialization topic.
+                println(s"`${thatScrutineeVar.name} is ${thatPattern}` is${if (split.isFallback) " " else " not "}fallback")
+                println(s"already removed = $full")
+                if (!split.isFallback && full && !head.continuation.isEmpty) {
+                  println(s"report warning")
+                  raiseDesugaringWarning(
+                    msg"found a duplicated case" -> thatPattern.toLoc,
+                    msg"the case is covered by pattern ${pattern.toString}" -> pattern.toLoc,
+                  )
+                }
+                specializeImpl(tail)(
+                  `-`(if (samePattern) continuation.isFull else full),
+                  scrutineeVar, scrutinee, pattern, context)
+              } else {
+                println(s"Case 1.2.2: $pattern are unrelated with $thatPattern")
```

Comment:
Sounds more natural:
```suggestion
                println(s"Case 1.2.2: $pattern are unrelated to $thatPattern")
```
https://textranch.com/c/unrelated-to-or-unrelated-with/